pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test: TestExecBuild_distsql_plan_locality_filter failed

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://mesolite.cluster.engflow.com/invocations/default/b53ff989-c5c7-41ed-a2d3-c3ab854a798c?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL29wdC9leGVjL2V4ZWNidWlsZGVyL3Rlc3RzL211bHRpcmVnaW9uLTlub2RlLTNyZWdpb24tM2F6czptdWx0aXJlZ2lvbi05bm9kZS0zcmVnaW9uLTNhenNfdGVzdA==) on master @ [f7f32d56c541ee6609580a0295e94a58750a261c](https://github.com/cockroachdb/cockroach/commits/f7f32d56c541ee6609580a0295e94a58750a261c):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestExecBuild_distsql_plan_locality_filter1487969987
    test_log_scope.go:82: use -show-logs to present logs inline
[15:12:54] --- done: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[15:12:55] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: outputs.zip/logTestExecBuild_distsql_plan_locality_filter1487969987
--- FAIL: TestExecBuild_distsql_plan_locality_filter (5.34s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colrpc.Inbox
            └ Node 7
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.51s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-63941

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21347989?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21347989?buildTab=artifacts#/) on master @ [ea000f02a90de0b07162e8ffafcfe3694c9f4bf6](https://github.com/cockroachdb/cockroach/commits/ea000f02a90de0b07162e8ffafcfe3694c9f4bf6):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter948361125
    test_log_scope.go:82: use -show-logs to present logs inline
[17:27:13] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/21181/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[17:27:13] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter948361125
--- FAIL: TestExecBuild_distsql_plan_locality_filter (4.09s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/21181/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/21181/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.14s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348329?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348329?buildTab=artifacts#/) on master @ [5ce700b6e247972c716c22a42361087b887ca8ef](https://github.com/cockroachdb/cockroach/commits/5ce700b6e247972c716c22a42361087b887ca8ef):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter2666079375
    test_log_scope.go:82: use -show-logs to present logs inline
[20:50:59] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19818/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[20:51:00] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter2666079375
--- FAIL: TestExecBuild_distsql_plan_locality_filter (5.52s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19818/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            ├ Node 8
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19818/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.56s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348444?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348444?buildTab=artifacts#/) on master @ [589d84273cafd97744e98579be3bf1e03801372f](https://github.com/cockroachdb/cockroach/commits/589d84273cafd97744e98579be3bf1e03801372f):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter3513103242
    test_log_scope.go:82: use -show-logs to present logs inline
[22:27:58] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/21179/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[22:27:58] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter3513103242
--- FAIL: TestExecBuild_distsql_plan_locality_filter (4.79s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/21179/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            ├ Node 8
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/21179/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.14s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348514?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348514?buildTab=artifacts#/) on master @ [c21e87c32f80e78f853e3f05bfb9785aac4f1575](https://github.com/cockroachdb/cockroach/commits/c21e87c32f80e78f853e3f05bfb9785aac4f1575):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter1424879769
    test_log_scope.go:82: use -show-logs to present logs inline
[22:38:32] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2627/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[22:38:32] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter1424879769
--- FAIL: TestExecBuild_distsql_plan_locality_filter (4.99s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2627/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            ├ Node 8
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/2627/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.16s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348539?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348539?buildTab=artifacts#/) on master @ [b0930f9cb4c4da2eebd23056fc2ee6792e20c6da](https://github.com/cockroachdb/cockroach/commits/b0930f9cb4c4da2eebd23056fc2ee6792e20c6da):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter113683198
    test_log_scope.go:82: use -show-logs to present logs inline
[22:49:43] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19405/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[22:49:44] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter113683198
--- FAIL: TestExecBuild_distsql_plan_locality_filter (9.39s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19405/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19405/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348621?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348621?buildTab=artifacts#/) on master @ [b0930f9cb4c4da2eebd23056fc2ee6792e20c6da](https://github.com/cockroachdb/cockroach/commits/b0930f9cb4c4da2eebd23056fc2ee6792e20c6da):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter4044038984
    test_log_scope.go:82: use -show-logs to present logs inline
[00:10:25] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1507/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[00:10:27] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter4044038984
--- FAIL: TestExecBuild_distsql_plan_locality_filter (8.80s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1507/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            ├ Node 8
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1507/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.12s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348639?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348639?buildTab=artifacts#/) on master @ [b0930f9cb4c4da2eebd23056fc2ee6792e20c6da](https://github.com/cockroachdb/cockroach/commits/b0930f9cb4c4da2eebd23056fc2ee6792e20c6da):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter944606871
    test_log_scope.go:82: use -show-logs to present logs inline
[00:24:20] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1527/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[00:24:20] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter944606871
--- FAIL: TestExecBuild_distsql_plan_locality_filter (12.82s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1527/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1527/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.12s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://mesolite.cluster.engflow.com/invocations/default/d0198380-b95d-45a5-98cb-14b390d2dd18?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL29wdC9leGVjL2V4ZWNidWlsZGVyL3Rlc3RzL211bHRpcmVnaW9uLTlub2RlLTNyZWdpb24tM2F6czptdWx0aXJlZ2lvbi05bm9kZS0zcmVnaW9uLTNhenNfdGVzdA==) on master @ [b0930f9cb4c4da2eebd23056fc2ee6792e20c6da](https://github.com/cockroachdb/cockroach/commits/b0930f9cb4c4da2eebd23056fc2ee6792e20c6da):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: outputs.zip/logTestExecBuild_distsql_plan_locality_filter650300587
    test_log_scope.go:82: use -show-logs to present logs inline
[04:18:46] --- done: /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[04:18:46] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: outputs.zip/logTestExecBuild_distsql_plan_locality_filter650300587
--- FAIL: TestExecBuild_distsql_plan_locality_filter (4.57s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /mnt/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.13s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348836?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21348836?buildTab=artifacts#/) on master @ [b0930f9cb4c4da2eebd23056fc2ee6792e20c6da](https://github.com/cockroachdb/cockroach/commits/b0930f9cb4c4da2eebd23056fc2ee6792e20c6da):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter1317863809
    test_log_scope.go:82: use -show-logs to present logs inline
[06:47:11] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19543/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[06:47:12] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
--- FAIL: TestExecBuild_distsql_plan_locality_filter (5.86s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19543/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            ├ Node 8
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19543/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.63s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21349092?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21349092?buildTab=artifacts#/) on master @ [b0930f9cb4c4da2eebd23056fc2ee6792e20c6da](https://github.com/cockroachdb/cockroach/commits/b0930f9cb4c4da2eebd23056fc2ee6792e20c6da):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter3548693747
    test_log_scope.go:82: use -show-logs to present logs inline
[08:16:52] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1507/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[08:16:52] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
*
* WARNING: disk slowness detected: unable to sync log files within 10s
*
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter3548693747
--- FAIL: TestExecBuild_distsql_plan_locality_filter (22.63s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1507/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colrpc.Inbox
            └ Node 7
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/1507/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.13s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21349343?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21349343?buildTab=artifacts#/) on master @ [6dc01a143d64b773bd2772fd232135501aef6c09](https://github.com/cockroachdb/cockroach/commits/6dc01a143d64b773bd2772fd232135501aef6c09):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter2761722876
    test_log_scope.go:82: use -show-logs to present logs inline
[15:32:15] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20437/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[15:32:16] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter2761722876
--- FAIL: TestExecBuild_distsql_plan_locality_filter (5.78s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20437/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            ├ Node 8
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20437/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.56s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21349354?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21349354?buildTab=artifacts#/) on master @ [3e041788939c2cfec43af8388d7b9524b85ea779](https://github.com/cockroachdb/cockroach/commits/3e041788939c2cfec43af8388d7b9524b85ea779):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter121973819
    test_log_scope.go:82: use -show-logs to present logs inline
[15:41:16] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20019/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[15:41:17] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter121973819
--- FAIL: TestExecBuild_distsql_plan_locality_filter (9.29s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20019/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colrpc.Inbox
            └ Node 7
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/20019/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.31s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/21348044?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/21348044?buildTab=artifacts#/) on master @ [b0930f9cb4c4da2eebd23056fc2ee6792e20c6da](https://github.com/cockroachdb/cockroach/commits/b0930f9cb4c4da2eebd23056fc2ee6792e20c6da):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/5b8474d924b8bcdae4b9dbded69996e3/logTestExecBuild_distsql_plan_locality_filter572790895
    test_log_scope.go:82: use -show-logs to present logs inline
[15:41:07] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15977/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[15:41:08] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/5b8474d924b8bcdae4b9dbded69996e3/logTestExecBuild_distsql_plan_locality_filter572790895
--- FAIL: TestExecBuild_distsql_plan_locality_filter (9.22s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15977/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/15977/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.38s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test.TestExecBuild_distsql_plan_locality_filter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21349379?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/21349379?buildTab=artifacts#/) on master @ [47b117c1c010ee4f61ce7a551cb06ce5b1108141](https://github.com/cockroachdb/cockroach/commits/47b117c1c010ee4f61ce7a551cb06ce5b1108141):


```
=== RUN   TestExecBuild_distsql_plan_locality_filter
    test_log_scope.go:172: test logs captured to: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter2057491479
    test_log_scope.go:82: use -show-logs to present logs inline
[16:33:55] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19894/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter with config multiregion-9node-3region-3azs: 9 tests, 2 failures
[16:33:56] --- total progress: 9 statements
--- total: 9 tests, 2 failures
    logic.go:5004: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/41a58169bb71c21e03d0883e4ca00029/logTestExecBuild_distsql_plan_locality_filter2057491479
--- FAIL: TestExecBuild_distsql_plan_locality_filter (6.62s)
=== RUN   TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement
    logic.go:3409: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19894/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:64: EXPLAIN (VEC) SELECT * FROM lf_distributed
        expected:
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 8
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
        but found (query options: "") :
            │
            ├ Node 1
            │ └ *colexec.ParallelUnorderedSynchronizer
            │   ├ *colrpc.Inbox
            │   └ *colrpc.Inbox
            ├ Node 7
            │ └ *colrpc.Outbox
            │   └ *colfetcher.ColBatchScan
            └ Node 9
              └ *colrpc.Outbox
                └ *colfetcher.ColBatchScan
            
    logic.go:2530: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/19894/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/multiregion-9node-3region-3azs/multiregion-9node-3region-3azs_test_/multiregion-9node-3region-3azs_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter:79: too many errors encountered, skipping the rest of the input
--- FAIL: TestExecBuild_distsql_plan_locality_filter/distributed_flow_placement (0.65s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_distsql_plan_locality_filter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter
+++ b/pkg/sql/opt/exec/execbuilder/testdata/distsql_plan_locality_filter
@@ -56,24 +56,22 @@ EXPLAIN (VEC) SELECT * FROM lf_distributed
 
 # With the filter set to us-east-1, scan flows should move off non-matching
 # nodes (1-6) onto us-east-1 nodes (7-9). The gateway (node 1) still
-# coordinates.
+# coordinates the result via Inboxes but does not run a local scan.
+#
+# We don't pin the EXPLAIN (VEC) output here because makeInstanceResolver
+# picks randomly among matching instances when the lease holder's locality
+# has no shared prefix with the eligible set (see distsql_physical_planner.go,
+# the SpanPartitionReason_LOCALITY_FILTERED_RANDOM branch). Instead, we
+# assert the deterministic invariant: nodes outside us-east-1 (specifically
+# nodes 2-6) must not appear anywhere in the plan.
 statement ok
 SET distsql_plan_locality_filter = 'region=us-east-1'
 
-query T
-EXPLAIN (VEC) SELECT * FROM lf_distributed
+query I
+SELECT count(*) FROM [EXPLAIN (VEC) SELECT * FROM lf_distributed]
+WHERE info ~ ' Node [2-6]$'
 ----
-│
-├ Node 1
-│ └ *colexec.ParallelUnorderedSynchronizer
-│   ├ *colrpc.Inbox
-│   └ *colrpc.Inbox
-├ Node 7
-│ └ *colrpc.Outbox
-│   └ *colfetcher.ColBatchScan
-└ Node 8
-  └ *colrpc.Outbox
-    └ *colfetcher.ColBatchScan
+0
 
 # Narrowing further to a specific AZ restricts to node 7 plus the gateway
 # (node 1) which coordinates the result.

